.gitlab/junit_upload/functional_test_junit_upload_system_probe.yml

```diff
@@ -23,22 +23,37 @@ functional_test_junit_upload_system_probe:
     - set -x
     - ss=0; for f in $DD_AGENT_TESTING_DIR/kitchen-junit-*.tar.gz; do [[ -e "$f" ]] || continue; inv -e junit-upload --tgz-path $f || ((ss++)); done; exit $ss
 
-functional_test_junit_upload_system_probe_kmt:
+.functional_test_junit_upload_system_probe_kmt:
   stage: functional_test_junit_upload
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/datadog-ci-uploader$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
   tags: ["arch:amd64"]
   allow_failure: true
   when: always
-  needs:
-    - job: kernel_matrix_testing_run_tests_x64
-      optional: true
-    - job: kernel_matrix_testing_run_tests_arm64
-      optional: true
   variables:
     DD_ENV: ci
   script:
     - set +x
     - export DATADOG_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.datadog_api_key_org2 --with-decryption --query "Parameter.Value" --out text)
     - export JIRA_TOKEN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.jira_read_api_token --with-decryption --query "Parameter.Value" --out text)
     - set -x
-    - ss=0; for f in $DD_AGENT_TESTING_DIR/junit-*.tar.gz; do [[ -e "$f" ]] || continue; inv -e junit-upload --tgz-path $f || ((ss++)); done; exit $ss
+    - ss=0; for f in $DD_AGENT_TESTING_DIR/junit-$ARCH*.tar.gz; do [[ -e "$f" ]] || continue; inv -e junit-upload --tgz-path $f || ((ss++)); done; exit $ss
+
+# junit upload jobs are separted because of gitlab limit on the number of jobs
+# a single job can 'need'.
+functional_test_junit_upload_system_probe_kmt_x64:
+  extends:
+    - .functional_test_junit_upload_system_probe_kmt
+  needs:
+    - job: kernel_matrix_testing_run_tests_x64
+      optional: true
+  variables:
+    ARCH: "x86_64"
```

Comment:
I don't think the arch-filtering is necessary, since it shouldn't include artifacts from the other jobs since they weren't `need`-ed.